### PR TITLE
[WPB-8712] use treefmt for everything

### PR DESCRIPTION
**Pull request by MangoIV**
https://wearezeta.atlassian.net/browse/WPB-8712
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Comment by MangoIV** [comment]
I think I want to readd some `make hlint-apply` rule or somth

File: changelog.d/5-internal/WPB-8712 [hunk 1]
```diff
@@ -0,0 +1 @@
+move formatting and linting of haskell files to treefmt, remove some of the now unneded rules
```
Comment (fisx) [inline changelog.d/5-internal/WPB-8712:1]:
```suggestion
move formatting and linting of haskell files to treefmt, remove some of the now unneeded rules
```

File: Makefile [hunk 2]
```diff
@@ -192,35 +179,27 @@ check-local-nix-derivations: regen-local-nix-derivations
 services: init install
 	$(MAKE) -C services/nginz
 
-# formats all Haskell files (which don't contain CPP)
+# formats everything according to treefmt rules
+# this may take a while (5 minutes) on first run but should be instant on 
+# any subsequent run
```
Comment (fisx) [inline Makefile:184]:
```suggestion
# any subsequent run except after you have changed files.
```

...  right?  this is nice, though.

File: Makefile [hunk 3]
```diff
@@ -129,9 +129,8 @@ devtest:
 .PHONY: sanitize-pr
 sanitize-pr:
 	./hack/bin/generate-local-nix-packages.sh
-	make formatf
 	make hlint-inplace-pr
-	make hlint-check-pr  # sometimes inplace has been observed not to do its job very well.
+	make format
```
Comment (fisx) [inline Makefile:133]:
```suggestion
	make format
	make formatc  # (at some point in the past, this has caught ormolu screwing up with inplace)
```

this *might* be fixed now with the `--check-inplace` argument, but can we be certain?

but if you prefer, we can remove this and see if it happens again.

**Comment by MangoIV** [comment]
there's one tradeoff to adding hlint to this is that it will make the initial lint very slow (will take ~5 minutes) but after then the cache kicks in and only changed files are re-linted.

**Review by fisx — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (MangoIV) [reply Makefile:133]: `--check-idempotence`? i have removed `formatc`, i think...

File: Makefile (hunk 2 above)
Comment (MangoIV) [reply Makefile:184]: yeah, depends on what instant means to you ;D

File: Makefile (hunk 3 above)
Comment (MangoIV) [reply Makefile:133]: i was hoping that ormolu by now is idempotent or at least the idempotency check works reliably :eyes:

**Pull request merged**